src/models/Scholarships.ts

```diff
@@ -155,6 +159,16 @@ class Scholarships extends FirestoreCollection<ScholarshipData> {
   ): FirestoreModel<ScholarshipData> {
     return new FirestoreModel<ScholarshipData>(this.collection.doc(), data);
   }
+
+  /**
+   *
+   * @param reqs is the scholarship requirements e.g. (grades, states, etc.)
+   * @param paramFilters
+   * @returns whether a scholarship's requirements are included in the given filters
+   */
+  includesFilter(reqs?: any[], paramFilters?: any[]): boolean {
```

Comment:
Hmm, I'm realizing this is a hard method to name because we want it to return true when:
- there are no requirements OR
- there are no filters for this requirement OR
- there is an intersection between the requirements and the filters

Ideally this could be named something like `containsAny` or `intersects` and its semantics are well understood but that would normally imply the first two cases are false rather than true.

Reply:
How about `requirementMatchesFilter`? You might want to give a meaningful JSDoc summary. E.g. 

```
Returns whether the given requirement list matches the given filter list. True is returned when any of the following conditions are met:
- There are no requirements.
- There are no filters.
- The requirement list intersects with the filter list.
```